src/librbd/io/ImageRequestWQ.cc

```diff
@@ -642,9 +642,10 @@ bool ImageRequestWQ<I>::needs_throttle(ImageDispatchSpec<I> *item) {
     throttle = t.second;
     tokens = item->tokens_requested(flag);
 
-    if (throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
-	      &ImageRequestWQ<I>::handle_throttle_ready>(
-		tokens, this, item, flag)) {
+    if (tokens &&
+          throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
+	    &ImageRequestWQ<I>::handle_throttle_ready>(
+	      tokens, this, item, flag)) {
```

Comment:
As it looks different in github about the indentation. I pasted the original code here, each indentation is two spaces. 
```
    if (tokens &&
          throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
            &ImageRequestWQ<I>::handle_throttle_ready>(
              tokens, this, item, flag)) {
      blocked = true;
    } else {
      item->set_throttled(flag);
    }
```